Extensions/XMPPMUCSUB/XMPPMUCSUB.h

```diff
@@ -0,0 +1,264 @@
+ * @see xmppMUCSUB:failedToUnsubscribeFromRoomJID:withError:
+ *
+ * @param roomJID the room that user wants to subscribe to.
+ * @param userJID Room moderator unsubscribes another room user.
+ */
+- (void)unsubscibeFromRoomJID:(XMPPJID *)roomJID userJID:(XMPPJID *)userJID;
+
+/**
+ * A user can query the MUC service to get their list of subscriptions.
+ *
+ */
+- (void)fetchSubscriptionList;
+
+/**
+ * A room moderator can get the list of subscribers by sending <subscriptions/> request directly to the room JID.
+ *
+ * @param roomJID the room that user wants to get list of subscribers.
+ */
+- (void)fetchSubscribersListForRoom:(XMPPJID *)roomJID;
+
+@end
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+#pragma mark -
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+@protocol XMPPMUCSUBDelegate
+@optional
+
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didReceiveMessage:(XMPPMessage *)message;
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didReceivePresence:(XMPPPresence *)message;
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didSubscribe:(NSXMLElement *)subscribe;
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didUnsubscribe:(NSXMLElement *)unsubscribe;
+
+/**
+ * Implement this method when calling [mucsubInstanse discoverFeatures]. It will be invoked if the request
+ * for discovering services is successfully executed and receives a successful response.
+ *
+ * @param sender XMPPMUCSUB object invoking this delegate method.
+ * @param features An array of NSXMLElements in the form shown below. You will need to extract the data you
+ *                 wish to use.
+ *
+ *                 <feature var="urn:xmpp:mucsub:0" />
+ */
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didDiscoverFeatures:(NSArray *)features;
+
+/**
+ * Implement this method when calling [mucsubInstanse discoverFeatures]. It will be invoked if the request
+ * for discovering services is unsuccessfully executed or receives an unsuccessful response.
+ *
+ * @param sender XMPPMUCSUB object invoking this delegate method.
+ * @param error NSError containing more details of the failure.
+ */
+- (void)xmppMUCSUBFailedToDiscoverFeatures:(XMPPMUCSUB *)sender withError:(NSError *)error;
+
+/**
+ * Implement this method when calling [mucsubInstanse discoverMUCSUBForRoomJID:]. It will be invoked if
+ * the request for discovering mucsub service is successfully executed and receives a successful response.
+ *
+ * @param sender XMPPMUCSUB object invoking this delegate method.
+ * @param features An array of NSXMLElements in the form shown below. You will need to extract the data you
+ *              wish to use.
+ *
+ *              <feature var='urn:xmpp:mucsub:0' />
+ *
+ * @param roomJID room JID that user wants to discover mucsub service for.
+ */
+- (void)xmppMUCSUB:(XMPPMUCSUB *)sender didDiscoverFeatures:(NSArray *)features ForRoomJID:(XMPPJID *)roomJID;
```

Comment:
Mixing camelCaseSyntax... Should probably be `forRoomJID`

Reply:
Done